lib/dataset/sized_dict.cpp

```diff
@@ -348,56 +318,237 @@ SizedDict<Key, Value>::merge_from(const SizedDict &other) const {
   return out;
 }
 
-template <class Key, class Value>
-bool SizedDict<Key, Value>::item_applies_to(const Key &key,
-                                            const Dimensions &dims) const {
+template <class Key, class Value, class Impl>
+bool SizedDict<Key, Value, Impl>::item_applies_to(
+    const Key &key, const Dimensions &dims) const {
   const auto &val = m_items.at(key);
   return std::all_of(val.dims().begin(), val.dims().end(),
                      [&dims](const Dim dim) { return dims.contains(dim); });
 }
 
-template <class Key, class Value>
-bool SizedDict<Key, Value>::is_edges(const Key &key,
-                                     const std::optional<Dim> dim) const {
+template <class Key, class Value, class Impl>
+bool SizedDict<Key, Value, Impl>::is_edges(const Key &key,
+                                           const std::optional<Dim> dim) const {
   const auto &val = this->at(key);
   return core::is_edges(m_sizes, val.dims(),
                         dim.has_value() ? *dim : val.dim());
 }
 
 template <class Key, class Value>
-core::Dict<Key, Value> union_(const SizedDict<Key, Value> &a,
-                              const SizedDict<Key, Value> &b,
-                              std::string_view opname) {
-  core::Dict<Key, Value> out;
+AlignedDict<Key, Value>::AlignedDict(
+    const Sizes &sizes,
+    std::initializer_list<std::pair<const Key, Value>> items,
+    const bool readonly)
+    : AlignedDict(sizes, raw_holder_type(items), readonly) {}
+
+template <class Key, class Value>
+AlignedDict<Key, Value>::AlignedDict(Sizes sizes, raw_holder_type items,
+                                     const bool readonly) {
+  this->m_sizes = std::move(sizes);
+  for (auto &&[key, value] : items)
+    set(key, std::move(value));
+  // `set` requires Dict to be writable, set readonly flag at the end.
+  this->m_readonly =
+      readonly; // NOLINT(cppcoreguidelines-prefer-member-initializer)
+}
+
+template <class Key, class Value>
+AlignedDict<Key, Value>::AlignedDict(Sizes sizes, holder_type items,
+                                     const bool readonly) {
+  this->m_sizes = std::move(sizes);
+  for (auto &&[key, value] : items)
+    set(key, std::move(value.value), value.aligned);
+  // `set` requires Dict to be writable, set readonly flag at the end.
+  this->m_readonly =
+      readonly; // NOLINT(cppcoreguidelines-prefer-member-initializer)
+}
+
+template <class Key, class Value>
+AlignedDict<Key, Value>::AlignedDict(
+    const SizedDict<Key, AlignedValue<Value>, AlignedDict<Key, Value>> &other)
+    : SizedDict<Key, AlignedValue<Value>, AlignedDict<Key, Value>>{other} {}
+
+template <class Key, class Value>
+AlignedDict<Key, Value>::AlignedDict(
+    SizedDict<Key, AlignedValue<Value>, AlignedDict<Key, Value>>
+        &&other) noexcept
+    : SizedDict<Key, AlignedValue<Value>, AlignedDict<Key, Value>>{
+          std::move(other)} {}
```

Comment:
Not default because...?

Reply:
Because this is not a regular move ctor. This takes a `SizedDict` as argument, not an `AlignedDict`.